qa/evil-tests/src/test/java/org/elasticsearch/index/engine/EvilInternalEngineTests.java

```diff
@@ -64,21 +64,6 @@ public void testOutOfMemoryErrorWhileMergingIsRethrownAndIsUncaught() throws IOE
                         public void merge(final MergePolicy.OneMerge merge) throws IOException {
                             throw new OutOfMemoryError("640K ought to be enough for anybody");
                         }
-
-                        @Override
-                        public synchronized MergePolicy.OneMerge getNextMerge() {
-                            /*
-                             * This will be called when we flush when we will not be ready to return the segments. After the segments are on
-                             * disk, we can only return them from here once or the merge scheduler will be stuck in a loop repeatedly
-                             * peeling off the same segments to schedule for merging.
-                             */
-                            if (segmentsReference.get() == null) {
-                                return super.getNextMerge();
-                            } else {
-                                final List<SegmentCommitInfo> segments = segmentsReference.getAndSet(null);
-                                return new MergePolicy.OneMerge(segments);
-                            }
```

Comment:
After removing this, the test times out. I have not found a way to do this in Lucene 8.6 as this is now private in the Index writer. @original-brownbear you were the last one to modify this class, you might have an idea what we can do here?

Reply:
The problem is that we need to somehow set a merge policy here that will actually trigger a background merge (the random policy used in this test triggers nothing which is kind of obvious when we only have a single document in the engine). 

=> As far as I understand it we need to craft a merge policy that will actually cause a background merge here I think